### PR TITLE
Add github action golangci-lint 

### PR DESCRIPTION
**Pull request by kim201212**
### 1. Type of change

Please delete options that are not relevant.

- [ ] New feature
- [x] Enhancement
- [ ] Bug/fix (non-breaking change which fixes an issue)
- [ ] others (anything other than above)

---

**Related:** #196

**Summary**

- Add github action for Golang lint

---

- ~@xellos00 Please add token named `GOLANG_LINT_ACTION_TOKEN`~
- Change token name from `GOLANG_LINT_ACTION_TOKEN`to `PR_ACTION_TOKEN`

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            go.mod
+            go.sum
+      - uses: golangci/golangci-lint-action@v3
+        with:
+          # Required: the version of golangci-lint is required and
+          # must be specified without patch version: we always use the
+          # latest patch version.
+          version: v1.49.0
+          args: --timeout 10m
+          github-token: ${{ secrets.PR_ACTION_TOKEN }}
```
Comment (rootwarp) [inline .github/workflows/golangci-lint.yml:39]: @kim201212 Would you mind explaining why we need `PR_ACTION_TOKEN` here, please?

Comment (kim201212) [reply .github/workflows/golangci-lint.yml:39]: It needs to run `env.GIT_DIFF`

Comment (rootwarp) [reply .github/workflows/golangci-lint.yml:39]: 😭

Comment (rootwarp) [reply .github/workflows/golangci-lint.yml:39]: Let's test without action token. I think it is not required.

Comment (kim201212) [reply .github/workflows/golangci-lint.yml]: https://github.com/dsrvlabs/vatz/pull/311/commits/7143eca01df2159d95a2620aadfff2a4ae5de469

I think the GitHub token is not required 😢

**Review by rootwarp — APPROVED** [review]
LGTM

**Review by xellos00 — APPROVED** [review]

**Pull request merged**